admin/src/api/post/controllers/post.js

```diff
@@ -29,13 +29,46 @@ module.exports = createCoreController("api::post.post", ({ strapi }) => ({
         },
       },
     });
+    entity.readingTime = getReadingTime(entity.content);
+
+    const recommandedPosts = await strapi.db.query("api::post.post").findMany({
+      select: ['title','published_on','tags','content','slug'],
+      where: {
+        $and: [
+          {
+            slug: { $ne: entity.slug },
+          },
+          {
+            is_resource: entity.is_resource,
+          },
+        ],
+      },
+      sort: { published_on: "desc" },
+      populate: {
+        image:true,
+        author: {
+          populate: {
+            image: true,
+          },
+        },
+      },
+    });
+    recommandedPosts.forEach(post =>  {
```

Comment:
You can do this in website, where we are setting other fields in `setPostFields` Method